### ISSUE TITLE
First Int is always 1

### ISSUE THREAD

**Issue by MrTarantula** [pomma89/Troschuetz.Random#6]
In the following scenario, `results` will always be `{ 0, 100000, 0, 0 }`. 
```C#
            var results = new int[4];
            for (int x = 0; x < 100000; x++)
            {
                var rnd = new TRandom();
                results[rnd.Next(4)]++;
            }
```

In the next scenario, `results` will be something like `{ 24971, 124881, 25068, 25080 }`.

```C#
            var results = new int[4];
            for (int x = 0; x < 100000; x++)
            {
                var rnd = new TRandom();
                results[rnd.Next(4)]++;
                results[rnd.Next(4)]++;
            }
```

So the first roll is always 1. The second is sufficiently random.

**Comment by WilStead** [comment]
I ran some additional tests and took a look at what was going on in the code. It's actually the first _and third_ results that are repeated, and it's not always 1: that result depends on the upper bound you've set.

The issue is a consequence of the way initialization has been implemented in the `XorShift128Generator` (the default, if you don't pick one). The first two results of the `NextULong` algorithm will always be fairly similar: roughly 8.2e18 first, then 7.5e18 second. The first and _third_ results of the `NextUIntInclusiveMaxValue` algorithm are similar for the same reason, with the skip over the second result being due to the way the generator caches half the bytes.

All the other random generation methods rely on the same algorithm, so the issue affects all of them. With a small range of possible values (like you'll get with `Next(4)`) those similar results get collapsed into the same value every time.

The issue could be resolved by changing the initialization of `_y ` to depend on the seed. `SeedY * seed` appears to produce sufficiently random results starting with the first call, though there may be more efficient options.

In the meantime, an easy workaround is to "prime" the generator by calling `Next()` three times after construction (and each call to `Reset`). The fourth result should always be sufficiently pseudo-random.
